File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int** sizes, size_t sizes_rows, size_t sizes_cols) {
    int max[2] = {0, 0};
    
    for(int i = 0; i < sizes_rows; i ++) {
        if(sizes[i][1] > sizes[i][0]) {
            int temp = sizes[i][0];
            sizes[i][0] = sizes[i][1];
            sizes[i][1] = temp;
        }
        
        if (sizes[i][0] > max[0]) max[0] = sizes[i][0];
        if (sizes[i][1] > max[1]) max[1] = sizes[i][1];
    }    
    
    return max[0] * max[1];
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    
    for i in range(len(sizes)):
        if sizes[i][1] > sizes[i][0]:
            sizes[i][0], sizes[i][1] = sizes[i][1], sizes[i][0]
    
    max0 = 0
    max1 = 0
    
    for i in range(len(sizes)):
        if sizes[i][0] > max0:
            max0 = sizes[i][0]
        if sizes[i][1] > max1:
            max1 = sizes[i][1]
            
    return max0 * max1